mongo/integration/unified_spec_test.go

```diff
@@ -447,15 +447,19 @@ func executeGridFSOperation(mt *mtest.T, bucket *gridfs.Bucket, op *operation) e
 }
 
 func executeTestRunnerOperation(mt *mtest.T, testCase *testCase, op *operation, sess mongo.Session) error {
+	if sess == nil {
+		return fmt.Errorf("received unexpected nil session")
+	}
+
 	var clientSession *session.Client
-	if sess != nil {
-		xsess, ok := sess.(mongo.XSession)
-		if !ok {
-			return fmt.Errorf("expected session type %T to implement mongo.XSession", sess)
-		}
-		clientSession = xsess.ClientSession()
+	xsess, ok := sess.(mongo.XSession)
+	if !ok {
+		return fmt.Errorf("expected session type %T to implement mongo.XSession", sess)
 	}
 
+	clientSession = xsess.ClientSession()
+	assert.NotNil(mt, clientSession, "expected valid session, got nil")
```

Comment:
Even though the ticket says "Would be good practice to add an assert.NotNil before each of these checks", it also says "We should just return errors so that the error chain is obvious in test debugging". Apologies for contradiction, we should stick with the latter.

Reply:
Done